File: Week Assignment 3 using Python/4_search_in_linked_list.py
# 4. Search in linked list
class Node:
    def __init__(self,data):
        self.data=data; self.next=None

def searchList(head, key):
    t = head
    while t:
        if t.data == key: return True
        t = t.next
    return False

if __name__ == "__main__":
    head = Node(5); head.next = Node(10)
    print(searchList(head, 10))
    print(searchList(head, 7))

File: Week Assignment 3 using C/4_search_in_linked_list.c
// 4. Search in linked list - return boolean (1 if found else 0)
#include <stdio.h>
#include <stdlib.h>

struct Node { int data; struct Node* next; };

int searchList(struct Node* head, int key) {
    while (head) {
        if (head->data == key) return 1;
        head = head->next;
    }
    return 0;
}

int main() {
    struct Node* head = (struct Node*)malloc(sizeof(struct Node)); head->data=5; head->next=NULL;
    head->next = (struct Node*)malloc(sizeof(struct Node)); head->next->data=10; head->next->next=NULL;
    printf("%d\n", searchList(head, 10)); // 1
    printf("%d\n", searchList(head, 7)); // 0
    return 0;
}
